flight/PiOS/STM32F4xx/inc/pios_dmashot.h

```diff
@@ -0,0 +1,78 @@
+#ifndef PIOS_DMASHOT_H
+#define PIOS_DMASHOT_H
+
+#include <stdint.h>
+
+#include <pios_stm32.h>
+#include "stm32f4xx_tim.h"
+
+#include "pios_tim_priv.h"
+
+
+#define DMASHOT_MAX_BUFFERS						8
+
+#define DMASHOT_MESSAGE_WIDTH					16
+#define DMASHOT_MESSAGE_PAUSE					2
+#define DMASHOT_STM32_BUFFER					(DMASHOT_MESSAGE_WIDTH+DMASHOT_MESSAGE_PAUSE)
+
+#ifdef STM32F446xx
+
+#define DSHOT150_FREQUENCY						11250000
+#define DSHOT300_FREQUENCY						22500000
+#define DSHOT600_FREQUENCY						45000000
+#define DSHOT900_FREQUENCY						60000000
+#define DSHOT1200_FREQUENCY						90000000
+
+#define DSHOT_TOTAL_PERIOD						74
+#define DSHOT_PERIOD_0							27
+#define DSHOT_PERIOD_1							55
+
+#else
+
+#define DSHOT300_FREQUENCY						6000000
+#define DSHOT600_FREQUENCY						(DSHOT300_FREQUENCY*2)
+#define DSHOT900_FREQUENCY						(DSHOT300_FREQUENCY*3)
+#define DSHOT1200_FREQUENCY						(DSHOT300_FREQUENCY*4)
+#define DSHOT150_FREQUENCY						(DSHOT300_FREQUENCY/2)
+
+#define DSHOT_TOTAL_PERIOD						19
+#define DSHOT_PERIOD_0							7
+#define DSHOT_PERIOD_1							14
+
+#endif
+
+struct pios_dmashot_timer_cfg {
+
+	TIM_TypeDef *timer;
+	DMA_Stream_TypeDef *stream;
+	uint32_t channel;
+	uint32_t tcif;
+
+};
+
+struct pios_dmashot_cfg {
+
+	const struct pios_dmashot_timer_cfg *timer_cfg;
+	uint8_t num_timers;
+
+};
+
+void PIOS_DMAShot_Init(const struct pios_dmashot_cfg *config);
```

Comment:
(Oh, I see they are in the C file.  Can you move them to the header, since this is the preferred way now?)